express-post-json/server.ts

```diff
@@ -1,39 +1,39 @@
 import express from 'express';
 
-type NextId = number;
 type Grade = {
   id: number;
   name: string;
   course: string;
   score: number;
 };
-type GradesArray = object[];
 
-let nextId: NextId = 1;
-const gradesArray: GradesArray = [];
+let nextId = 1;
+const grades: Record<number, Grade> = {};
+const gradesArray: Grade[] = [];
```

Comment:
This works, but it requires you to write code below that keeps `grades` and `gradesArray` in sync. That's very likely to lead to bugs as people enhance and modify the code because they'll probably forget to do that.

There is no real benefit in doing it this way. If you convert `grades` to a `gradesArray` in your `app.get` route handler, which is the only place you need to do it, it will add possibly a millisecond to your execution time (and probably less).

Computers are very, very fast these days. You do not want to do "premature optimization", which is a term we use to describe the tendency to optimize code to run as fast as possible before we've actually proven that such optimization is necessary.

You may not have had execution performance in mind when you did it this way, though! But hopefully either the maintenance discussion or the performance discussion cover enough to convince you to not do it this way. :-)